src/main/java/games/strategy/triplea/delegate/PurchaseDelegate.java

```diff
@@ -300,16 +300,9 @@ public String purchaseRepair(final Map<Unit, IntegerMap<RepairRule>> repairRules
     return repairMap;
   }
 
-  Comparator<RepairRule> repairRuleComparator = new Comparator<RepairRule>() {
-    UnitTypeComparator utc = new UnitTypeComparator();
-
-    @Override
-    public int compare(final RepairRule o1, final RepairRule o2) {
-      final UnitType u1 = (UnitType) o1.getResults().keySet().iterator().next();
-      final UnitType u2 = (UnitType) o2.getResults().keySet().iterator().next();
-      return utc.compare(u1, u2);
-    }
-  };
+  Comparator<RepairRule> repairRuleComparator = Comparator.comparing(
+      (final RepairRule o) -> (UnitType) o.getResults().keySet().iterator().next(),
+      new UnitTypeComparator());
```

Comment:
Looks like the `repairRuleComparator` field could be made `private static final` without issue.

Also, is it necessary to qualify the type of the key extractor function parameter (i.e. `final RepairRule o`)?  Is there some kind of ambiguity, or is it just for readability?

Reply:
It's neccessary to either define the type that way, or to call Comparator#comparing this way: `Comparator.<RepairRule, UnitType>comparing(...)` since the cast is neccessary either way, this way is less code and equally readable

Reply:
Weird.  I wonder why my IDE shows that the parameter type is inferred correctly when it is removed:

![type-inference](https://user-images.githubusercontent.com/4826349/33801279-36d2a2f0-dd24-11e7-8ddc-37b9598b61bf.png)